src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/BundleSlingInitialContentJarEntryExtractor.java

```diff
@@ -0,0 +1,146 @@
+        this.contentReaderProvider = contentReaderProvider;
+        this.parentFolderRepoInitHandler = parentFolderRepoInitHandler;
+    }
+
+    /**
+     *
+     * @return {@code true} in case the given entry was part of the initial content otherwise {@code false}
+     * @throws Exception
+     */
+    public void extractSlingInitialContent(@NotNull BundleSlingInitialContentExtractorContext context, 
+                                           @NotNull SlingInitialContentBundleEntryMetaData slingInitialContentBundleEntryMetaData, 
+                                           @NotNull Set<SlingInitialContentBundleEntryMetaData> collectedSlingInitialContentBundleEntries) throws IOException, ConverterException {
+
+        String repositoryPath = slingInitialContentBundleEntryMetaData.getRepositoryPath();
+        File file = slingInitialContentBundleEntryMetaData.getTargetFile();
+        PathEntry pathEntryValue = slingInitialContentBundleEntryMetaData.getPathEntry();
+        // all entry paths used by entry handlers start with "/"
+        String contentPackageEntryPath = "/" + org.apache.jackrabbit.vault.util.Constants.ROOT_DIR + PlatformNameFormat.getPlatformPath(repositoryPath);
+
+        Path tmpDocViewInputFile = null;
+
+        try(InputStream bundleFileInputStream = new FileInputStream(file)) {
+            VaultPackageAssembler packageAssembler = assemblerProvider.initPackageAssemblerForPath(context, repositoryPath, pathEntryValue);
+
+            final ContentReader contentReader = contentReaderProvider.getContentReaderForEntry(file, pathEntryValue);
+            if (contentReader != null) {
+
+                // convert to docview xml
+                tmpDocViewInputFile = Files.createTempFile(context.getConverter().getTempDirectory().toPath(), "docview", ".xml");
+                try (OutputStream docViewOutput = Files.newOutputStream(tmpDocViewInputFile, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING)) {
+
+                    repositoryPath = FilenameUtils.removeExtension(repositoryPath);
+                    boolean isFileDescriptorEntry = isFileDescriptor(collectedSlingInitialContentBundleEntries, contentPackageEntryPath);
+                    VaultContentXMLContentCreator contentCreator = new VaultContentXMLContentCreator(StringUtils.substringBeforeLast(repositoryPath, "/"), docViewOutput, context.getNamespaceRegistry(), packageAssembler, isFileDescriptorEntry);
+
+
+                    if(file.getName().endsWith(".xml")){
+                        contentCreator.setIsXmlProcessed();
+                    }
+
+                    contentReader.parse(file.toURI().toURL(), contentCreator);
+                    contentPackageEntryPath =  new ContentPackageEntryPathComputer(collectedSlingInitialContentBundleEntries, contentPackageEntryPath, contentCreator).compute();
+                    contentCreator.finish();
+
+                } catch (IOException | XMLStreamException e) {
```

Comment:
XMLStreamException is never thrown in the corresponding block and could be removed (also from the list of imports)